src/executor.ts

```diff
@@ -5,20 +5,19 @@ import {getVersionToUse} from "./version-parser";
 
 const scopeAgentMetadataURL = "https://repo1.maven.org/maven2/com/undefinedlabs/scope/scope-agent/maven-metadata.xml";
 const scopeMavenInstrMetadataURL = "https://repo1.maven.org/maven2/com/undefinedlabs/scope/scope-instrumentation-for-maven/maven-metadata.xml";
+const scopeNoTrackDep = "_scope_notrackdep";
 
 export async function instrument(allowBeta:boolean): Promise<void> {
     const workdir = process.cwd();
     const [agentVersion, instrVersion] = await Promise.all([getVersionToUse(scopeAgentMetadataURL, allowBeta), getVersionToUse(scopeMavenInstrMetadataURL, false)])
 
     const scopeAgentPath = await tc.downloadTool(`https://repo1.maven.org/maven2/com/undefinedlabs/scope/scope-agent/${agentVersion}/scope-agent-${agentVersion}.jar`);
-    if(!scopeAgentPath.endsWith(".jar")) {
-        await io.mv(scopeAgentPath, scopeAgentPath+".jar");
-    }
+    const finalScopeAgentPath = (!scopeAgentPath.endsWith(".jar")) ? scopeAgentPath + scopeNoTrackDep + ".jar" : scopeAgentPath.replace(".jar", scopeNoTrackDep+".jar");
```

Comment:
I think this is a bit too much for one line

```suggestion
   let finalScopeAgentPath
   if (!scopeAgentPath.endsWith(".jar")) {
     finalScopeAgentPath = `${scopeAgentPath}${scopeNoTrackDep}.jar`
   else {
     finalScopeAgentPath = scopeAgentPath.replace('.jar', `${scopeNoTrackDep}.jar`)
   }
```

and I think the logic can be simplified as well, if you can assume there will be no `.jar` in the middle of the string:
```suggestion
const finalScopeAgentPath = `${scopeAgentPath.replace('.jar', '')}${scopeNoTrackDep}.jar`
```